File: src/main/java/com/github/theonlytails/rubymod/registries/BiomeRegistry.java
package com.github.theonlytails.rubymod.registries;

import com.github.theonlytails.rubymod.RubyMod;
import com.github.theonlytails.rubymod.world.biomes.RubyHillsBiome;
import net.minecraft.world.biome.Biome;
import net.minecraftforge.common.BiomeDictionary;
import net.minecraftforge.common.BiomeManager;
import net.minecraftforge.fml.RegistryObject;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;

public class BiomeRegistry {
    public static final DeferredRegister<Biome> BIOMES = DeferredRegister.create(ForgeRegistries.BIOMES, RubyMod.MOD_ID);

    public static final RegistryObject<Biome> RUBY_HILLS = BIOMES.register("ruby_hills", RubyHillsBiome::new);

    public static void registerBiomes() {
        registerBiome(RUBY_HILLS.get(), BiomeDictionary.Type.PLAINS, BiomeDictionary.Type.OVERWORLD);
    }

    public static void registerBiome(Biome biome, BiomeDictionary.Type... types) {
        BiomeManager.addBiome(BiomeManager.BiomeType.COOL, new BiomeManager.BiomeEntry(biome, 100));
        BiomeDictionary.addTypes(biome, types);
        BiomeManager.addSpawnBiome(biome);
    }
}

File: src/main/kotlin/com/github/theonlytails/rubymod/registries/BiomeRegistry.kt
package com.github.theonlytails.rubymod.registries

import com.github.theonlytails.rubymod.RubyMod
import com.github.theonlytails.rubymod.world.BiomeMaker
import net.minecraft.util.RegistryKey
import net.minecraft.util.registry.Registry
import net.minecraftforge.common.BiomeManager
import net.minecraftforge.event.world.BiomeLoadingEvent
import net.minecraftforge.registries.ForgeRegistries
import thedarkcolour.kotlinforforge.forge.KDeferredRegister

/**
 * Registers and adds custom biomes to the world.
 *
 * @author TheOnlyTails
 */
object BiomeRegistry {
	val BIOMES = KDeferredRegister(ForgeRegistries.BIOMES, RubyMod.MOD_ID)

	private val RUBY_HILLS by BIOMES.registerObject("ruby_hills", BiomeMaker::makeRubyHills)
	private val rubyHillsRegistryKey = RegistryKey.getOrCreateKey(
		Registry.BIOME_KEY, RubyMod.id("ruby_hills"))

	fun biomeLoading(event: BiomeLoadingEvent) {
		if (event.name == RUBY_HILLS.registryName) {
			BiomeManager.addBiome(BiomeManager.BiomeType.WARM,
				BiomeManager.BiomeEntry(rubyHillsRegistryKey, 6))
		}
	}
}